fuse_core/src/uuid.cpp

```diff
@@ -88,23 +88,21 @@ UUID generate()
 {
   static std::random_device rd;
   static std::mt19937 generator(rd());
-  static std::uniform_int_distribution<uint32_t> distibution;
-
-  UUID u;
+  static std::uniform_int_distribution<uint64_t> distibution;
+  static std::mutex distribution_mutex;
 
-  int i = 0;
-  auto random_value = distibution(generator);
-  for (UUID::iterator it = u.begin(); it != u.end(); ++it, ++i)
+  uint64_t random1;
```

Comment:
Please pay extra attention to this refactor.

My interpretation of the original code is that a random number of N bytes is generated. Inside the loop, each random byte is copied into the next open slot in the UUID. When you run out of random bytes, generate N more bytes.

The refactor relies on the uint8_t type in the UUID and the uint64_t type generated by the random number generator being the expected sizes.

That seems like a strong guarantee in C++:
>unsigned integer type with width of exactly 8, 16, 32 and 64 bits respectively
(provided only if the implementation directly supports the type)

https://en.cppreference.com/w/cpp/types/integer